### PR TITLE
Typing improvements to dnautils and sam.bwa_mem

### PR DESCRIPTION
**Pull request by nh13**

### CODE REVIEW

**Comment by nh13** [comment]
@jtratner it would be great to a release after this PR.    The `  _BASE_QUALITY_OFFSET: ClassVar[int] = 33` change specifically is a bugfix as I believe `attr` needs the type annotation work.

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        PYTHON_VERSION: [3.6.10, 3.7.6, 3.8.2]
+        PYTHON_VERSION: [3.6.10, 3.7.7, 3.8.2]
```
Comment (jtratner) [inline .github/workflows/pythonpackage.yml]: Better to just not specify it and instead use latest of each? (Ie 3.6/3.7/3.8 and no minor version$

**Review by jtratner — APPROVED** [review]

**Comment by jtratner** [comment]
After this merges happy to cut new release (we just have to tag the commit on master and then run the internal gocd pipeline on that commit)

**Comment by nh13** [comment]
@jtratner this and #7 would be great to get in before a release.  Happy to help, just need some guidance on running the internal gocd pipeline (which one, any config, etc.)

**Review by pkaleta — APPROVED** [review]
LGTM

**Comment by jtratner** [comment]
Steps to cut release:

1. Merge PRs (ideally with release notes identifying version+ changes, optional to also do `poetry version XYZ`
2.  Tag latest commit on master with a version
3. Run `samwell-build` internally

**Comment by jtratner** [comment]
can you add a really short release note in docs/release_notes.rst

```
v0.1.1
------
Add in some typing to XYZ
```

**Pull request merged**